### PR TITLE
RMSprop update policy implementation

### PR DESCRIPTION
**Pull request by ivmarkp**
Implementation of the RMSprop optimizer using the RMSpropUpdate policy class. All tests pass locally. Please review, thanks.

### CODE REVIEW

File: src/mlpack/core/optimizers/rmsprop/rmsprop_update.hpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ */
+class RMSpropUpdate
+{
+ public:
+  /**
+   * Construct the RMSprop update policy with the given epsilon parameter.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
```
Comment (zoq) [inline src/mlpack/core/optimizers/rmsprop/rmsprop_update.hpp]: The smoothing parameter is called alpha.

File: src/mlpack/core/optimizers/rmsprop/rmsprop_update.hpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  RMSpropUpdate(const double epsilon = 1e-8,
+             const double alpha = 0.99) :
```
Comment (zoq) [inline src/mlpack/core/optimizers/rmsprop/rmsprop_update.hpp]: Minor style issue; line-up alpha with epsilon.

File: src/mlpack/core/optimizers/rmsprop/rmsprop.hpp [hunk 3]
```diff
@@ -86,7 +90,7 @@ class RMSprop
   RMSprop(DecomposableFunctionType& function,
```
Comment (zoq) [inline src/mlpack/core/optimizers/rmsprop/rmsprop.hpp]: Minor style issue (it was wrong in the first place); line-up parameters with DecomposableFunctionType.

File: src/mlpack/core/optimizers/rmsprop/rmsprop.hpp [hunk 4]
```diff
@@ -2,8 +2,9 @@
  * @file rmsprop.hpp
  * @author Ryan Curtin
  * @author Marcus Edel
+ * @author Vivek Pal
  *
- * RMSprop optimizer. RmsProp is an optimizer that utilizes the magnitude of
+ * RMSprop optimizer. RMSprop is an optimizer that utilizes the magnitude of
```
Comment (rcurtin) [inline src/mlpack/core/optimizers/rmsprop/rmsprop.hpp]: I know this is picky, but should we call it `RMSProp` not `RMSprop` like in Hinton's lectures?

Comment (ivmarkp) [inline src/mlpack/core/optimizers/rmsprop/rmsprop.hpp]: We should probably call it `RMSProp`. The reason I changed it to `RMSprop` was to make it consistent with the rest of the code. Should I go ahead and do the renaming?

Comment (zoq) [inline src/mlpack/core/optimizers/rmsprop/rmsprop.hpp]: Yes, please, people probably heard of RMSProp or read about it in some paper and might be confused if we use a different name.

Comment (ivmarkp) [inline src/mlpack/core/optimizers/rmsprop/rmsprop.hpp]: Okay, making the changes now.

**Pull request merged**